plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/type/TestTypeInfoUtils.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.bigquery.type;
+
+import org.assertj.core.api.ObjectAssert;
+import org.junit.jupiter.api.Test;
+
+import static io.trino.plugin.bigquery.type.TypeInfoUtils.getTypeInfoFromTypeString;
+import static io.trino.plugin.bigquery.type.TypeInfoUtils.getTypeInfosFromTypeString;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestTypeInfoUtils
```

Comment:
I think we can move some tests to TestBigQueryType and remove redundant tests.

Reply:
Most of the negative tests here verify the parser doesn't throw out of range errors. These tests are cheap, and I'd keep them here for completeness. I'll add tests for all supported types, and special chars in `BaseBigQueryConnectorTest`.